test/39-compose-images.spec.ts

```diff
@@ -0,0 +1,530 @@
+import * as _ from 'lodash';
+
+import { docker } from '../src/lib/docker-utils';
+import { expect } from './lib/chai-config';
+import * as Images from '../src/compose/images';
+import * as mockedDockerode from './lib/mocked-dockerode';
+import * as mockedDatabase from './lib/mocked-database';
+import * as db from '../src/db';
+
+describe('compose/images', () => {
+	before(() => {
+		mockedDatabase.create();
+	});
+
+	after(() => {
+		try {
+			mockedDatabase.restore();
+		} catch (e) {
+			/* noop */
+		}
+	});
+
+	afterEach(() => {
+		// Clear Dockerode actions recorded for each test
+		mockedDockerode.resetHistory();
+	});
+
+	it('Removes a legacy Image', async () => {
```

Comment:
I modified this crafted these "legacy" payloads from real data since I've not seen legacy images but I'm used the legacy logic to make these.